### PR TITLE
Pal 702 updated Jenkinsfile to checkout develop branch

### PR DESCRIPTION
**Pull request by pd104923**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -165,7 +165,7 @@ spec:
 
         stage('Install, Unit Tests, Checkstyle') {
             dir('Palisade-services') {
-                git url: 'https://github.com/gchq/Palisade-services.git'
+                git branch: 'develop', url: 'https://github.com/gchq/Palisade-services.git'
                 sh "git checkout ${GIT_BRANCH_NAME}"
```
Comment (developer01189998819991197253) [inline Jenkinsfile]:
```suggestion
                git branch: GIT_BRANCH_NAME, url: 'https://github.com/gchq/Palisade-services.git'
```

File: Jenkinsfile [hunk 2]
```diff
@@ -181,7 +181,7 @@ spec:
             // If this branch name exists in integration-tests, use that
             // Otherwise, default to integration-tests/develop
             dir('Palisade-integration-tests') {
-                git url: 'https://github.com/gchq/Palisade-integration-tests.git'
+                git branch: 'develop', url: 'https://github.com/gchq/Palisade-integration-tests.git'
                 sh "git checkout ${GIT_BRANCH_NAME} || git checkout develop"
```
Comment (developer01189998819991197253) [inline Jenkinsfile]: It'll first checkout develop, go to checkout GIT_BRANCH_NAME but if none exists, it'll still be on develop and then will still build, therefore you can remove the or
```suggestion
                sh "git checkout ${GIT_BRANCH_NAME}"
```

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]
Can be done on one line

**Review by m78233 — DISMISSED** [review]

**Review by developer01189998819991197253 — DISMISSED** [review]

File: Jenkinsfile [hunk 3]
```diff
@@ -182,8 +181,8 @@ spec:
             // If this branch name exists in integration-tests, use that
             // Otherwise, default to integration-tests/develop
             dir('Palisade-integration-tests') {
-                git url: 'https://github.com/gchq/Palisade-integration-tests.git'
-                sh "git checkout ${GIT_BRANCH_NAME} || git checkout develop"
+                git branch: 'develop', url: 'https://github.com/gchq/Palisade-integration-tests.git'
+                sh "git checkout ${GIT_BRANCH_NAME}"
```
Comment (ac74475) [inline Jenkinsfile]:
```suggestion
                sh(script: "git checkout ${GIT_BRANCH_NAME}", returnStatus: true)
```

**Review by developer01189998819991197253 — APPROVED** [review]

**Pull request merged**